pkg/lib/config/oauth.go

```diff
@@ -212,17 +231,44 @@ type OAuthClientConfig struct {
 	CustomUIURI                            string                       `json:"x_custom_ui_uri,omitempty"`
 	App2appEnabled                         bool                         `json:"x_app2app_enabled,omitempty"`
 	App2appInsecureDeviceKeyBindingEnabled bool                         `json:"x_app2app_insecure_device_key_binding_enabled,omitempty"`
-	AuthenticationFlowGroupAllowlist       *[]string                    `json:"x_authentication_flow_group_allowlist,omitempty"`
 	AuthenticationFlowAllowlist            *AuthenticationFlowAllowlist `json:"x_authentication_flow_allowlist,omitempty"`
 }
 
 type AuthenticationFlowAllowlist struct {
-	SignupFlows          []string `json:"signup_flows"`
-	PromoteFlows         []string `json:"promote_flows"`
-	LoginFlows           []string `json:"login_flows"`
-	SignupLoginFlows     []string `json:"signup_login_flows"`
-	ReauthFlows          []string `json:"reauth_flows"`
-	AccountRecoveryFlows []string `json:"account_recovery_flows"`
+	Groups *[]string                         `json:"groups,omitempty"`
+	Flows  *AuthenticationFlowAllowlistFlows `json:"flows,omitempty"`
+}
+
+type AuthenticationFlowAllowlistFlows []AuthenticationFlowAllowlistFlow
+
+func (flows AuthenticationFlowAllowlistFlows) FilterByType(t AuthenticationFlowAllowlistFlowType) []string {
+	var filtered []string
+	for _, flow := range flows {
+		if flow.Type == t {
+			filtered = append(filtered, flow.Name)
+		}
+	}
+	return filtered
+}
+
+type AuthenticationFlowAllowlistFlow struct {
+	Type AuthenticationFlowAllowlistFlowType `json:"type"`
+	Name string                              `json:"name"`
+}
+
+type AuthenticationFlowAllowlistFlowType string
+
+const (
+	AuthenticationFlowAllowlistFlowTypeSignup          AuthenticationFlowAllowlistFlowType = "signup"
+	AuthenticationFlowAllowlistFlowTypePromote         AuthenticationFlowAllowlistFlowType = "promote"
+	AuthenticationFlowAllowlistFlowTypeLogin           AuthenticationFlowAllowlistFlowType = "login"
+	AuthenticationFlowAllowlistFlowTypeSignupLogin     AuthenticationFlowAllowlistFlowType = "signup_login"
+	AuthenticationFlowAllowlistFlowTypeReauth          AuthenticationFlowAllowlistFlowType = "reauth"
+	AuthenticationFlowAllowlistFlowTypeAccountRecovery AuthenticationFlowAllowlistFlowType = "account_recovery"
+)
+
+func (flow AuthenticationFlowAllowlistFlow) Compare(other AuthenticationFlowAllowlistFlow) bool {
```

Comment:
We do not need this method. Struct in Golang is equal to each other if all fields are equal.